cmds/webboot/webboot.go

```diff
@@ -66,11 +78,50 @@ func (d *DownloadByLink) Exec() error {
 	return nil
 }
 
+// Exec displays subdirectory or cached isos under the path directory
+func (g *DirGroup) Exec() error {
+
+	entries := []menu.Entry{}
+
+	readerInfos, err := ioutil.ReadDir(g.path)
+	if err != nil {
+		return err
+	}
+
+	// check the directory, if there is a subdirectory, add another DirGroup option
+	// if there is iso file, add a chacedIso option
+	for _, info := range readerInfos {
+		if info.IsDir() {
+			entries = append(entries, &DirGroup{
+				name:     info.Name(),
+				path:     filepath.Join(g.path, info.Name()),
+				uiEvents: g.uiEvents,
+			})
+		} else if filepath.Ext(info.Name()) == ".iso" {
+			iso := &CachedISO{
+				path:  filepath.Join(g.path, info.Name()),
+				label: info.Name(),
+			}
+			entries = append(entries, iso)
+		}
+	}
+	_, err = menu.DisplayMenu("Groups", "Choose a group of chached iso", entries, g.uiEvents)
+	return err
+}
+
+// Exec displays bookmarks under a certain group
+func (g *BookmarkGroup) Exec() error {
+	_, err := menu.DisplayMenu("Distro", "Choose a Distro", g.entries, g.uiEvents)
+	return err
+}
+
+// linkOpen sets up a http session to the URL
 func linkOpen(URL string) (io.ReadCloser, error) {
 	u, err := url.Parse(URL)
 	if err != nil {
 		log.Fatal(err)
 	}
+	// if the link is not begin with "file", "https" and "http", we do not support it
```

Comment:
nit: I think this comment actually isnt needed because by looking at the error message on link 138, someone would understand this already